src/stream.rs

```diff
@@ -25,28 +25,32 @@ impl<'a> WrappedSlice<'a> {
     self.offset
   }
 
-  fn consume(&mut self, n_bytes: usize) -> usize {
+  fn consume(&mut self, n_bytes: usize) -> Option<usize> {
     let offset = self.offset.saturating_add(n_bytes);
     // Make sure `offset <= buf.len()`
-    self.offset = std::cmp::min(self.buf.len(), offset);
-    self.offset
+    if offset <= self.buf.len() {
+      self.offset = offset;
+      Some(offset)
+    } else {
+      None
+    }
   }
 
-  fn read_into(&mut self, out_buffer: &mut [u8]) -> usize {
+  fn read_into(&mut self, out_buffer: &mut [u8]) -> Option<usize> {
     // Get number of remaining bytes.
     let remaining = self.remaining();
     if remaining == 0 {
       // No more bytes.
-      return 0;
+      return None;
     }
 
     // Try to fill the output buffer.
     let n_read = std::cmp::min(remaining, out_buffer.len());
     let offset = self.offset;
-    let end_off = self.consume(n_read);
+    let end_off = self.consume(n_read).unwrap();
```

Comment:
unwrap would panic.  Better to just return `None`.

```suggestion
    let end_off = self.consume(n_read)?;
```

Reply:
That call shouldn't return `None` there, but I would prefer to avoid `unwarp()`.

Reply:
`n_read` is limited to `remaining`, which ensures that the call will never fail (if it fails, it is a bug in the api).